### ISSUE TITLE
ClientCookieDecoder doesn't support some valid RFC6265 expires fields

### ISSUE THREAD

**Issue by slandelle** [netty/netty#6016]
`ClientCookieDecoder` currently uses `HttpHeaderDateFormat` underneath, which has a fallback strategy and tries the following patterns:
* `E, dd MMM yyyy HH:mm:ss z`
* `EEEE, d-MMM-y HH:mm:ss z`
* `EEE, MMM d HH:mm:ss yyyy`

The former is more or less RFC1123 (it accepts full days of week, while RFC1123 requires 3 letters), I don't know where the 2 others come from.
RFC1123 is [the spec for headers such as `Expires` or `Date`](https://www.w3.org/Protocols/rfc2616/rfc2616-sec14.html).

The problem is that RFC6265 defines [its own spec](https://tools.ietf.org/html/rfc6265#section-5.1.1) (shaking head) that supports RFC1123 format.

While most people stick to RFC1123, some don't.
For example, Github uses a thing of beauty such as `Sun, 27 Nov 2016 19:37:15 -0000` (`-0000` instead of `GMT` (banging head on desk).
Basically, the spirit is that:
* the day of week is ignored (completely redundant/useless)
* the timezone is ignored as RFC6265 enforces UTC
* delimiters can be many things, so it's not problem to write something like `27-Nov-2016`

Working on a fix.

**Comment by slandelle** [comment]
Will send a PR tomorrow.

Side effect of custom impl:
```
Benchmark                    Mode  Cnt        Score        Error  Units
ParserBenchmark.parseNetty  thrpt   20   442613,292 ±  26450,297  ops/s
ParserBenchmark.parseNew    thrpt   20  4335260,828 ± 296953,584  ops/s
```

**Comment by maseev** [comment]
@slandelle Can you tell me what tool you used to build such a benchmark?

**Comment by normanmaurer** [comment]
@maseev that is jmh:  http://openjdk.java.net/projects/code-tools/jmh/

**Comment by maseev** [comment]
@normanmaurer nice! thanks!

**Comment by slandelle** [comment]
PR done: #6016

**Comment by normanmaurer** [comment]
Fixed by https://github.com/netty/netty/pull/6018
